src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/WriteMavenPomStep.java

```diff
@@ -141,8 +150,14 @@ protected Void run() throws Exception {
             if (path.isDirectory()) {
                 throw new FileNotFoundException(path.getRemote() + " is a directory.");
             }
+
+            final Model model = step.getModel();
+            if (model == null) {
+                throw new AbortException("The specified Maven Model is null. Probably Pipeline has been restarted, " +
```

Comment:
No, this is impossible, because we are using `SynchronousNonBlockingStepExecution`.